spec/controllers/v0/sessions_controller_spec.rb

```diff
@@ -80,14 +97,38 @@
         expect(User.find(uuid)).to_not be_nil
         expect(User.find(uuid).attributes).to eq(User.from_merged_attrs(loa1_user, loa3_user).attributes)
       end
-      context ' when SAMLResponse is invalid' do
-        before { allow(OneLogin::RubySaml::Response).to receive(:new).and_return(invalid_saml_response) }
+      context ' when user clicked DENY' do
+        before { allow(OneLogin::RubySaml::Response).to receive(:new).and_return(saml_response_click_deny) }
+        it 'redirects to an auth failure page' do
+          expect(Rails.logger).to receive(:warn).with(/#{SAML::AuthFailHandler::CLICKED_DENY_MSG}/)
+          expect(post(:saml_callback)).to redirect_to(SAML_CONFIG['relay'] + '?auth=fail')
```

Comment:
Expectations of a redirect to `?auth=fail` and a status code of `:found` is repeated in 4 examples... not very DRY.  Anyone have suggestions to DRY this up?